mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1043,6 +1043,60 @@ struct ReorderElementwiseOpsOnBroadcast final
   }
 };
 
+/// Pattern to rewrite a ExtractOp(Elementwise) -> Elementwise(ExtractOp).
+/// This may result in more efficient code when we extracting a single value
+/// from multi-element vector and also to help canonicalize 1-element vectors to
+/// scalars.
```

Comment:
I am wondering about this ... The tests for the vectorizer became "cleaner" (nice!), but if a value is already in a vector register, and then it's extracted ... I would actually expect the original code to be faster. However, the overall performance will depend on the surrounding code 🤔 

I am mostly waving my hands here, but perhaps refrain from saying that this "may result in more efficient code" and instead say "replaces elementwise computation on N vectors with an elementwise on M elements (M < N) and an additional `vector.extract`". Basically, something a bit more neutral.

Reply:
changed to "cleaner" )